### PR TITLE
Nwd nnw

### PR DESCRIPTION
**Pull request by Regggis**
Last homework done - every numbers found ;)

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: nwd-nnw
🏅 Granted 6 XP for @Regggis

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Regggis for on-time delivery

File: homework/nwd-nnw/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,40 @@
 #pragma once
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
+
+    if(lhs==0 && rhs==0){return 0;}
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:5]: Add space between operations like '+' '-' '!=' '==' etc...

File: homework/nwd-nnw/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+
+    int divider;
+
+    if (lhs>rhs){divider = lhs;}
+    else {divider = rhs;}
+
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:11]: Too many new lines

File: homework/nwd-nnw/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,40 @@
 #pragma once
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
+
+    if(lhs==0 && rhs==0){return 0;}
+
+    int divider;
+
+    if (lhs>rhs){divider = lhs;}
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:9]: Add spaces

File: homework/nwd-nnw/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
+    else {divider = rhs;}
+
+    if(divider<0){divider=divider*(-1);}
+    while(lhs%divider!=0 || rhs%divider!=0)
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:13]: Add spaces

File: homework/nwd-nnw/nwdNww.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:5]: if (condition) {
 sth....
}

File: homework/nwd-nnw/nwdNww.hpp (hunk 3 above)
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:9]: if (condition) {
 sth....
}

Also use std::swap instead create new variable

if (lhs > rhs) {
   std::swap(lhs, rhs)
}

File: homework/nwd-nnw/nwdNww.hpp [hunk 5]
```diff
@@ -1,11 +1,40 @@
...
+    return divider;
+    
     return -1;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
+    int multiple;
+
+    if(lhs ==0 || rhs==0){return 0;}
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:26]: every command in new line

File: homework/nwd-nnw/nwdNww.hpp [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
+
+    if (rhs>lhs){multiple = rhs;}
+    else{multiple = lhs;}
+
+    if(multiple<0){multiple=multiple*(-1);}
+
+    while(multiple%lhs!=0 || multiple%rhs!=0)
+    {
+        ++multiple;
+    }
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:36]: add spaces

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**